ACCS-366: My Account: Stored payment methods page (#816)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/user-account/functions.mdx

| [`getCountries`](#getcountries) | Calls the countries query to retrieve a list of countries. |
| [`getCustomer`](#getcustomer) | Retrieves the customer information for the logged-in customer. |
| [`getCustomerAddress`](#getcustomeraddress) | Returns an array of addresses associated with the current customer. |

> | [`getCustomerPaymentTokens`](#getcustomerpaymenttokens) | Retrieves stored payment tokens for the logged-in customer via the customerPaymentTokens query. |

Reviewer: ```suggestion
| [`getCustomerPaymentTokens`](#getcustomerpaymenttokens) | Retrieves stored payment tokens for the logged-in customer via the `customerPaymentTokens query`. |
```

> | [`deletePaymentToken`](#deletepaymenttoken) | Removes a stored payment token using the deletePaymentToken mutation. |

Reviewer: ```suggestion
| [deletePaymentToken](#deletepaymenttoken) | Removes a stored payment token using the `deletePaymentToken mutation`. |
```

See [`CustomerAddressesModel`](#customeraddressesmodel).

## getCustomerPaymentTokens

> The `getCustomerPaymentTokens` function loads saved payment methods (vault tokens) for the logged-in customer. It calls the <Link href="https://developer.adobe.com/commerce/webapi/graphql/schema/checkout/queries/customer-payment-tokens/" text="customerPaymentTokens" /> query over the drop-in GraphQL client (GET, no cache). Results are transformed into display rows for the **PaymentMethods** container.

Reviewer: ```suggestion
The `getCustomerPaymentTokens` function retrieves saved payment methods (vault tokens) for the logged-in customer. It calls the <Link href="https://developer.adobe.com/commerce/webapi/graphql/schema/checkout/queries/customer-payment-tokens/" text="`customerPaymentTokens`" /> query via the drop-in GraphQL client (GET, no cache), then transforms the results into display rows for the `PaymentMethods` container.
```

```ts
const getCustomerPaymentTokens = async (
  filterPaymentMethodCodes?: string[]
): Promise<StoredPaymentMethodDisplay[]>
```

<TableWrapper nowrap={[0, 1]}>

| Parameter | Type | Req? | Description |
| --- | --- | --- | --- |
| `filterPaymentMethodCodes` | `string[]` | No | When set, only tokens whose `payment_method_code` equals or starts with one of these strings are returned. |

</TableWrapper>

### Events

> Does not emit any drop-in events. To **populate the PaymentMethods list** from client-side data without calling this query first, emit **account/customerPaymentTokens** on the event bus (see [Events](/dropins/user-account/events/#accountcustomerpaymenttokens-listens)). That event is for **display data only**; removing a token uses **`deletePaymentToken`**, not the bus.

Reviewer: ```suggestion
Does not emit any drop-in events. To populate the `PaymentMethods` list from client-side data without calling this query, emit `account/customerPaymentTokens` on the event bus (see [Events](/dropins/user-account/events/#accountcustomerpaymenttokens-listens)). This event is for display data only. Token removal still uses `deletePaymentToken`, not the bus.
```

### Returns

Returns [`StoredPaymentMethodDisplay[]`](#storedpaymentmethoddisplay).

## deletePaymentToken

The `deletePaymentToken` function removes a stored payment token for the logged-in customer. It calls the <Link href="https://developer.adobe.com/commerce/webapi/graphql/schema/checkout/mutations/delete-payment-token/" text="deletePaymentToken" /> mutation with the token `public_hash`.

> The built-in **[PaymentMethods](/dropins/user-account/containers/payment-methods/)** container calls this **only after** the shopper confirms removal in the confirmation dialog. If you call `deletePaymentToken` from your own code, the mutation runs **immediately**—there is no confirmation UI unless you implement it.

Reviewer: ```suggestion
The built-in [PaymentMethods](/dropins/user-account/containers/payment-methods/) container calls this **only after** the shopper confirms removal in the confirmation dialog. If you call `deletePaymentToken` from your own code, the mutation runs immediately and there is no confirmation UI unless you implement it.
```

}
```

### StoredPaymentMethodDisplay

> The `StoredPaymentMethodDisplay` object is returned as list items by [`getCustomerPaymentTokens`](#getcustomerpaymenttokens). The same shape is the typed payload for the **`account/customerPaymentTokens`** event when you inject a list on the bus (see [Events](/dropins/user-account/events/#typescript-declaration)). It matches the **PaymentMethods** / **PaymentCard** view model (plus `publicHash` for keys and for **`deletePaymentToken`**).

Reviewer: ```suggestion
The StoredPaymentMethodDisplay object is returned as list items by [`getCustomerPaymentTokens`](#getcustomerpaymenttokens). The same shape is also used as the typed payload for the `account/customerPaymentTokens` event when injecting a list via the event bus (see [Events](/dropins/user-account/events/#typescript-declaration)). It aligns with the `PaymentMethods` / `PaymentCard` view model and includes `publicHash` for keys and for use with `deletePaymentToken`.
```
